File: backend/master/src/main/java/com/ociet/loadgenerator/master/Producer.java
package com.ociet.loadgenerator.master;

import com.ociet.loadgenerator.common.Constants;
import com.ociet.loadgenerator.common.LoadRequestMessage;
import com.ociet.loadgenerator.common.LoadSubrequestMessage;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
public class Producer {
    private static final Logger logger = LoggerFactory.getLogger(Producer.class);

    @Autowired
    private KafkaTemplate<String, LoadSubrequestMessage> kafkaSubrequestTemplate;

    @Autowired
    private KafkaTemplate<String, LoadRequestMessage> kafkaRequestTemplate;

    public void sendSubrequest(String key, int partition, LoadSubrequestMessage message) {
        logger.info(String.format("Producing message -> %s", message));
        this.kafkaSubrequestTemplate.send(Constants.LOAD_SUBREQUEST_TOPIC, partition, key, message);
    }

    public void sendRequest(String key, LoadRequestMessage message) {
        logger.info(String.format("Producing message -> %s", message));
        this.kafkaRequestTemplate.send(Constants.LOAD_REQUEST_TOPIC, key, message);
    }
}

File: backend/master/src/main/kotlin/com/chylb/loadgenerator/master/Producer.kt
package com.chylb.loadgenerator.master

import com.chylb.loadgenerator.common.Constants
import com.chylb.loadgenerator.common.LoadRequestMessage
import com.chylb.loadgenerator.common.LoadSubrequestMessage
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service


@Service
class Producer {
    companion object {
        private val logger: Logger = LoggerFactory.getLogger(Producer::class.java)
    }

    @Autowired
    private val kafkaSubrequestTemplate: KafkaTemplate<String, LoadSubrequestMessage?>? = null

    @Autowired
    private val kafkaRequestTemplate: KafkaTemplate<String, LoadRequestMessage?>? = null

    @Autowired
    private val kafkaCancelTemplate: KafkaTemplate<String, String>? = null

    fun sendSubrequest(key: String, partition: Int, message: LoadSubrequestMessage?) {
        logger.info(String.format("Producing message -> %s", message))
        kafkaSubrequestTemplate!!.send(Constants.LOAD_SUBREQUEST_TOPIC, partition, key, message)
    }

    fun sendRequest(key: String, message: LoadRequestMessage?) {
        logger.info(String.format("Producing message -> %s", message))
        kafkaRequestTemplate!!.send(Constants.LOAD_REQUEST_TOPIC, key, message)
    }

    fun sendCancel() {
        logger.info(String.format("Producing cancel message "))
        kafkaCancelTemplate!!.send(Constants.LOAD_CANCEL_TOPIC, "")
    }
}